parlai/scripts/train_model.py

```diff
@@ -426,11 +512,27 @@ def _average_dicts(self, all_versions):
                 finalized[k] = np.mean(values)
         return finalized
 
+    def _cleanup_inaccurate_metrics(self, metrics):
+        """
+        Remove inaccurate multiworld metrics.
+
+        When training in multitask mode, agent-level metrics may be shown, but are actually averages
+        not distinguished across the worlds. This method hides those tasks.
+
+        Issue: https://github.com/facebookresearch/ParlAI/issues/1750
+        """
+        # TODO: fix the root issue
+        if 'tasks' in metrics:
+            metrics[
+                'warning'
+            ] = 'agent level metrics (e.g. loss, mean_loss, ppl) are averaged over tasks'
+
     def _sync_training_metrics(self, metrics):
         """
         Sync training metrics across workers. A handful of special cases are handled
         as exceptions, and the remaining metrics are simply averaged across workers.
         """
+        self._cleanup_inaccurate_metrics(metrics)
```

Comment:
Only remaining change is to make this not called from _sync_training_metrics, and call it in the log() function instead.

Reply:
moved